pilot/pkg/features/pilot.go

```diff
@@ -254,6 +254,13 @@ var (
 			"legacy deployments that have not yet been migrated to the new safe regular expressions.",
 	)
 
+	EnableEndpointSliceController = env.RegisterBoolVar(
+		"PILOT_ENABLE_ENDPOINT_SLICE_CONTROLLER",
+		false,
+		"If enabled, Pilot will use EndpointSlices as the source of endpoints for Kubernetes services. "+
```

Comment:
may be state its mutually exclusive? if not, also state so

Reply:
Its mutually exclusive in this iteration. I think there is some value in making it not behave this way, but don't want to bloat the PR, making both work concurrently is strictly additional work anyways.

I'll update the message

Reply:
+1 mutually exclusive sounds good for now